cmd/args_kubernetes.go

```diff
@@ -385,25 +385,11 @@ func init() {
 	}
 	{
 		const (
-			key          = keys.K8SCoreDNSMetricsPort
-			longOpt      = "k8s-core-dns-metrics-port"
-			envVar       = release.ENVPREFIX + "_K8S_CORE_DNS_METRICS_PORT"
-			description  = "CoreDNS metrics port if annotations not on service definition"
-			defaultValue = defaults.K8SCoreDNSMetricsPort
-		)
-
-		rootCmd.PersistentFlags().String(longOpt, defaultValue, envDescription(description, envVar))
-		bindFlagError(longOpt, viper.BindPFlag(key, rootCmd.PersistentFlags().Lookup(longOpt)))
-		bindEnvError(envVar, viper.BindEnv(key, envVar))
-		viper.SetDefault(key, defaultValue)
-	}
-	{
-		const (
-			key          = keys.K8SKubeDNSMetricsPort
-			longOpt      = "k8s-kube-dns-metrics-port"
-			envVar       = release.ENVPREFIX + "_K8S_KUBE_DNS_METRICS_PORT"
-			description  = "Kube dns metrics port if annotations not on service definition"
-			defaultValue = defaults.K8SKubeDNSMetricsPort
+			key          = keys.K8SDNSMetricsPort
```

Comment:
Not suggesting a change here, just observing this wild way of setting up command line flags processing, with these scope blocks and shadowed const values. I would think a slice of objects containing the various values in the const data and a loop funning the Cobra and Viper commands, would result in less code. Again, not suggesting you change this in this or any PR, just observing that it is interesting.

Reply:
Agreed, it's untenable and makes for difficult changes to the config structure.